upup/pkg/fi/cloudup/bootstrapchannelbuilder.go

```diff
@@ -70,11 +71,26 @@ func (b *BootstrapChannelBuilder) buildManifest() (*channelsapi.Addons, map[stri
 		Manifest: fi.String("dns-controller/v1.4.1.yaml"),
 	})
 
+	// TODO: Create configuration object for cni providers (maybe create it but orphan it)?
 	if b.cluster.Spec.Networking.Kopeio != nil {
 		key := "networking.kope.io"
 		version := "1.0.20161116"
 
-		// TODO: Create configuration object (maybe create it but orphan it)?
+		location := key + "/v" + version + ".yaml"
+
+		addons.Spec.Addons = append(addons.Spec.Addons, &channelsapi.AddonSpec{
+			Name:     fi.String(key),
+			Version:  fi.String(version),
+			Selector: map[string]string{"k8s-addon": key},
+			Manifest: fi.String(location),
+		})
+
+		manifests[key] = "addons/" + location
+	}
+
+	if b.cluster.Spec.Networking.Weave != nil {
```

Comment:
The code here is asking 

We are 3 layers deep into possibly nil struct pointers here.. can we add a check before we assume this is good?

```
if cluster.Spec != nil && cluster.Spec.Networking != nil && cluster.Spec.Networking.Weave != nil 
```

I would like for us to get in the habit of this - because I have noticed a few things fall through the cracks when using kops as a library..